sdk/eventhub/azure-eventhub/azure/eventhub/aio/_consumer_async.py

```diff
@@ -162,7 +165,8 @@ async def _open_with_retry(self) -> None:
         await self._do_retryable_operation(self._open, operation_need_param=False)
 
     async def _message_received(self, message: Message) -> None:
-        self._message_buffer.append(message)
+        async with self._message_buffer_lock:
+            self._message_buffer.append(message)
```

Comment:
What does the lock around this achieve?
We're no longer processing the incoming message batch concurrently with `gather` - so shouldn't that remove the possibility of a race condition?

Reply:
we have two coroutines interacting with self._message_buffer simultaneously -- receive task keeps adding messages while callback task keeps pumping messages, race condition between the two tasks is what I'm trying to avoid here.

Reply:
We could investigate this further in term of optimization - it seem expensive to await the lock for every individual message.

Reply:
yeah.. it is interesting that actually in my v0, I indeed tried to batch things, however, I guess I didn't do thing correctly that the perf is not good.

I can put a todo for this